File: 3254. Find the Power of K-Size Subarrays I/Solution.py
from typing import List


class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        ans = [-1] * (len(nums) - k + 1)
        windows_size = 0
        for i, x in enumerate(nums):
            if i == 0 or x == nums[i - 1] + 1:
                windows_size += 1
            else:
                windows_size = 1
            if windows_size >= k:
                ans[i - k + 1] = x
        return ans

File: 3254. Find the Power of K-Size Subarrays I/Solution.c
#include <stdlib.h>
#include <string.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* resultsArray(int* nums, int numsSize, int k, int* returnSize)
{
    *returnSize = numsSize - k + 1;
    int* ans = (int*)malloc(sizeof(int) * (*returnSize));
    memset(ans, -1, sizeof(int) * (*returnSize));

    int window_size = 0;
    for (int i = 0; i < numsSize; ++i) {
        if (i == 0 || nums[i] == nums[i - 1] + 1)
            ++window_size;
        else
            window_size = 1;

        if (window_size >= k)
            ans[i - k + 1] = nums[i];
    }
    return ans;
}
